Update docs to detail unsupported Tiered Storage functionality for 5.4 [HZ-4381] (#1034)
hazelcast/hz-docs: docs/modules/tiered-storage/pages/overview.adoc

== Not Yet Supported

- Data recovery from persisted data: When a Hazelcast member has Tiered Storage enabled and it fails or shuts down, you are not able to use the local device files of Tiered Storage for data recovery. Data recovery will be supported in the future releases.

> - TTL expiration is not supported by Tiered-Storage-backed maps: methods like `IMap.put` will throw `UnsupportedOperationException` if a TTL value is provided. If the default TTL setting is present in the map configuration, `InvalidConfigurationException` will be thrown during the node startup.

Reviewer: ```suggestion
- TTL expiration is not supported by Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a TTL value is provided. If the default TTL setting is present in the map configuration, `InvalidConfigurationException` is thrown during the node startup.
```

> - TTL expiration is not supported by Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a TTL value is provided. If the default TTL setting is present in the map configuration, `InvalidConfigurationException` is thrown during the node startup.

Reviewer 2: ```suggestion
- Time-to-live (TTL) expiration is not supported by the Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a TTL value is provided. If the default TTL setting is present in the map configuration, `InvalidConfigurationException` is thrown during the member startup.
```

> - Max-Idle expiration is not supported by Tiered-Storage-backed maps: methods like `IMap.put` will throw `UnsupportedOperationException` if a Max-Idle value is provided. If the default Max-Idle setting is present in the map configuration, `InvalidConfigurationException` will be thrown during the node startup.

Reviewer: ```suggestion
- Max-Idle expiration is not supported by Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a Max-Idle value is provided. If the default Max-Idle setting is present in the map configuration, `InvalidConfigurationException` is thrown during the node startup.
```

> - Max-Idle expiration is not supported by Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a Max-Idle value is provided. If the default Max-Idle setting is present in the map configuration, `InvalidConfigurationException` is thrown during the node startup.

Reviewer 2: ```suggestion
- Max-Idle expiration is not supported by the Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a Max-Idle value is provided. If the default Max-Idle setting is present in the map configuration, `InvalidConfigurationException` is thrown during the member startup.
```

> - Eviction is not supported: if an eviction policy is configured for a Tiered-Storage-backed map, `InvalidConfigurationException` will be thrown during the node startup.

Reviewer: ```suggestion
- Eviction is not supported. If an eviction policy is configured for a Tiered-Storage-backed map, `InvalidConfigurationException` is thrown during the node startup.
```

> - Eviction is not supported. If an eviction policy is configured for a Tiered-Storage-backed map, `InvalidConfigurationException` is thrown during the node startup.

Reviewer 2: ```suggestion
- Eviction is not supported. If an eviction policy is configured for a Tiered-Storage-backed map, `InvalidConfigurationException` is thrown during the member startup.
```

> - Data Persistence and Tiered Storage are mutually exclusive features: if both are enabled simultaneously, `InvalidConfigurationException` will be thrown during the node startup.

Reviewer: ```suggestion
- Data Persistence and Tiered Storage are mutually exclusive features. If both are enabled simultaneously, `InvalidConfigurationException` is thrown during the node startup.
```

> - Data Persistence and Tiered Storage are mutually exclusive features. If both are enabled simultaneously, `InvalidConfigurationException` is thrown during the node startup.

Reviewer 2: ```suggestion
- Data Persistence and Tiered Storage are mutually exclusive features. If both are enabled simultaneously, `InvalidConfigurationException` is thrown during the member startup.
```

> - SQL is not supported for Tiered-Storage-backed maps: `UnsupportedOperationException` is thrown if SQL query is being executed. Use Predicate API instead.

Reviewer: ```suggestion
- SQL is not supported for Tiered-Storage-backed maps. `UnsupportedOperationException` is thrown when SQL query is executed. To avoid this, use Predicate API instead.
```
